sympy/integrals/integrals.py

```diff
@@ -715,8 +734,29 @@ def eval_factored(f, x, a, b):
                                        for f in integrals])
                         try:
                             evalued = Add(*others)._eval_interval(x, a, b)
-                            evalued_pw = piecewise_fold(Add(*piecewises))._eval_interval(x, a, b)
-                            function = uneval + evalued + evalued_pw
+                            # Handles periodic functions by adjusting
+                            # integration limits and splitting the interval
+                            # into periodic segments and a remainder. Evaluates
+                            # the integral over one period and the
+                            # remainder, combining these results with proper
+                            # handling of integration direction.
+                            period = None
+                            if not isinstance(self.function, Poly)  and x.kind is NumberKind:
+                                period = periodicity(self.function, x)
```

Comment:
I don't think that calling periodicity here is the right fix. I think that the bug is in `_eval_interval` and should be fixed there in the first instance rather than adding additional code here to work around it.

Reply:
the actual issue may be  here:
```python
# In the file `solvers/inequalities.py`, lines 509-511
if sup - inf is S.Infinity:
    domain = Interval(0, period, False, True).intersect(_domain)
    _domain = domain
```
As mentioned in the above code, the piecewise function is splitting only in (0 , period) which is causing incorrect value for the inegrals whose limits are outside (0,period) so i am using this formula ∫(0 to b+nT) f(x) dx = ∫(0 to b) f(x) dx + n ∫(0 to T) f(x) dx to calucalate the integral in the limits (0,period) so that the above splitting won't cause any issue. so here i need to change the code in doit function as i need to transform the function such that limits are changed to (0,b).